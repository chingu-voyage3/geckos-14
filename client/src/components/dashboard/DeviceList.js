/** DeviceList Component :
The DeviceList displays  a list of Devices using a simple CSS grid
TODO: Allow hiding the List
 */

import React from 'react';
import Device from './Device';

const renderDevice = (device, actions) => {
  // console.log('renderDevice', device);
  return (
    <div
      className="device"
      key={device.name}
      onClick={() => {
        actions.toogleSelectedDevice(device.id);
      }}
    >
      <Device {...device} />
    </div>
  );
};
const renderDeviceList = props => {
  // console.log('renderDeviceList');
  return props.devices.map(device => {
    return renderDevice(device, props.actions);
  });
};
const DeviceList = props => {
  // console.log('DeviceList Component');
  if (props.devices.length === 0) return <div className="loading">Loading Data...</div>;
  return <div className="device-list">{renderDeviceList(props)}</div>;
};

export default DeviceList;
