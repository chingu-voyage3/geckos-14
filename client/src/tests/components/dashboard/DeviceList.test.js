import React from 'react';
import DeviceList from '../../../components/dashboard/DeviceList';

import { shallow } from 'enzyme';
const devices = [
  { name: 'device1', status: 'online', type: 'Thermometer' },
  { name: 'device2', status: 'offline', type: 'PhMeter' },
  { name: 'device3', status: 'offline', type: 'Thermometer' }
];
it('renders without crashing', () => {
  const wrapper = shallow(<DeviceList devices={devices} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
