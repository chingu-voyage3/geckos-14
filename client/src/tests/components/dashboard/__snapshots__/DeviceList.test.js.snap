// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="device-list"
>
  <div
    className="device"
    key="dev0"
    onClick={[Function]}
  >
    <Device
      id="dev0"
      name="LondonTemp"
      url="http://devices.webofthings.io/pi/sensors/temperature/"
    />
  </div>
  <div
    className="device"
    key="dev1"
    onClick={[Function]}
  >
    <Device
      id="dev1"
      name="LondonHumidity"
      url="http://devices.webofthings.io/pi/sensors/humidity/"
    />
  </div>
</div>
`;
